runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -13165,7 +13165,8 @@ VMgenerateCatchBlockBBStartPrologue(
       TR::Register * biAddrReg = cg->allocateRegister();
 
       // Load address of counter into biAddrReg
-      genLoadAddressConstant(cg, node, (uintptr_t) comp->getRecompilationInfo()->getCounterAddress(), biAddrReg);
+      generateRegLitRefInstruction(cg, TR::InstOpCode::getLoadOpCode(), node, biAddrReg,
+                                   (uintptr_t)comp->getRecompilationInfo()->getCounterAddress(), TR_BodyInfoAddress);
```

Comment:
Is this going to generate suboptimal code in non-AOT compiles?

Reply:
Also why are you using `TR_BodyInfoAddress` instead of `TR_CatchBlockCounter`?

Reply:
I'm not sure about the generated code. I can add a test to use `genLoadAddressConstant` in the non-AOT case. Though now that I look at `genLoadAddressConstant`, it should generate the relocation record automatically as long as

```c++
   if (node->getOpCode().hasSymbolReference())
      sym = node->getSymbol();
```

is true of the `node` that we pass to `genLoadAddressConstant` (with the associated symbol being the recompilation counter symref), so maybe setting that somehow would be better than using the lower-level function.